astronomer/providers/sftp/hooks/sftp.py

```diff
@@ -0,0 +1,127 @@
+from fnmatch import fnmatch
+from typing import List
+
+import asyncssh
+from airflow.exceptions import AirflowException
+from airflow.hooks.base import BaseHook
+from asgiref.sync import sync_to_async
+
+
+class SFTPHookAsync(BaseHook):
+    """
+    Interact with an SFTP server via asyncssh package
+
+    :param sftp_conn_id: SFTP connection ID to be used for connecting to SFTP server
+    :param host: hostname of the SFTP server
+    :param port: port of the SFTP server
+    :param username: username used when authenticating to the SFTP server
+    :param password: password used when authenticating to the SFTP server
+                     Can be left blank if using a key file
+    :param known_hosts: path to the known_hosts file on the local file system
+                     If known_hosts is set to the literal "none", then no host verification is performed
+    :param key_file: path to the client key file used for authentication to SFTP server
+    :param passphrase: passphrase used with the key_file for authentication to SFTP server
+    """
+
+    conn_name_attr = "ssh_conn_id"
+    default_conn_name = "sftp_default"
+    conn_type = "sftp"
+    hook_name = "SFTP"
+    default_known_hosts = '"~/.ssh/known_hosts"'
+
+    def __init__(  # nosec: B107
+        self,
+        sftp_conn_id: str = default_conn_name,
+        host: str = "",
+        port: int = 22,
+        username: str = "",
+        password: str = "",
+        known_hosts: str = default_known_hosts,
+        key_file: str = "",
+        passphrase: str = "",
+    ) -> None:
+        self.sftp_conn_id = sftp_conn_id
+        self.host = host
+        self.port = port
+        self.username = username
+        self.password = password
+        self.known_hosts = known_hosts
+        self.key_file = key_file
+        self.passphrase = passphrase
+
+    async def _get_conn(self) -> asyncssh.SSHClientConnection:
+        """
+        Asynchronously connect to the SFTP server as an SSH client
+
+        The following parameters are provided either in the extra json object in
+        the SFTP connection definition
+
+        key_file
+        known_hosts
+        passphrase
```

Comment:
shall we keep these points as bullet points, how is it looking in docs currently?